UPGRADE.md

```diff
@@ -0,0 +1,7 @@
+This document describes future changes that will infect your current resources used inside of OpenShift.
+Each change contains description of the change and a date when that change will happen.
+
+* Build's pod label will be renamed from `build` to `io.openshift.build.name`.
+  All the newly created Builds will already have the new label, but old ones will
+  be recognized (inside the build controller) until post 3.1/1.2 version.
```

Comment:
"old values will be recognized by OpenShift through the 3.1 release.  Support for these values may be removed in any major point release after 3.1 (e.g. 3.2)"

@smarterclayton please confirm that's the version guarantee we want to make here.